lowcar/devices/KoalaBear/KoalaBear.cpp

```diff
@@ -365,24 +353,24 @@ int KoalaBear::sign (float x) {
  * Make sure that at least one of the pins is set to 255 at all times.
  */
 void KoalaBear::drive (float target, uint8_t mtr) {
-	int pin1 = (mtr == MTRA) ? AIN1 : BIN1; //select the correct pins based on the motor
-	int pin2 = (mtr == MTRA) ? AIN2 : BIN2;
-
-	int direction = sign(target);
-	int currpwm1 = 255;
-	int currpwm2 = 255;
-
-	// Determine how much to move the pins down from 255
-	// Sanity check: If |target| == 1, move max speed --> pwm_difference == 255
-	//				 If |target| == 0, stop moving	  --> pwm_difference == 0
-	unsigned int pwm_difference = (target * direction) * 255 ; // A number between 0 and 255 inclusive (255 is stop; 0 is max speed);
-
-	if (direction > 0) { // Moving forwards
-		currpwm1 -= pwm_difference;
-	} else if (direction < 0) { // Moving backwards
-		currpwm2 -= pwm_difference;
-	}
-
-	analogWrite(pin1, currpwm1);
-	analogWrite(pin2, currpwm2);
+    int pin1 = (mtr == MTRA) ? AIN1 : BIN1; // select the correct pins based on the motor
+    int pin2 = (mtr == MTRA) ? AIN2 : BIN2;
+
+    float direction = (target > 0.0) ? 1.0 : -1.0;
+    int currpwm1 = 255;
+    int currpwm2 = 255;
+
+    // Determine how much to move the pins down from 255
+    // Sanity check: If |target| == 1, move max speed --> pwm_difference == 255
+    //				 If |target| == 0, stop moving	  --> pwm_difference == 0
+    int pwm_difference = (int)(target * direction * 255.0); // A number between 0 and 255 inclusive (255 is stop; 0 is max speed);
+
+    if (direction > 0) { // Moving forwards
+        currpwm1 -= pwm_difference;
```

Comment:
why isn't this +=

Reply:
It has to do with the controls for the motors... basically if both control pins are high, the motor is stopped; lowering the voltage on one pin while keeping the other one high moves the motor one way (and the lower the voltage is, the faster it goes); lowering the voltage on the other pin while keeping the first pin high moves the motor the other way.